##토큰화와 벡터화 같은 기본적인 NLP 전처리를 마친 후 pycaret 적용



1.   BOW-TF (Countvectorizer)
2.   BOW-TFIDF (TFIDFvectorizer) ☑





In [1]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [2]:
pip show packaging

Name: packaging
Version: 21.3
Summary: Core utilities for Python packages
Home-page: https://github.com/pypa/packaging
Author: Donald Stufft and individual contributors
Author-email: donald@stufft.io
License: BSD-2-Clause or Apache-2.0
Location: /usr/local/lib/python3.10/dist-packages
Requires: pyparsing
Required-by: arviz, astropy, bokeh, build, dask, db-dtypes, distributed, fastai, fastcore, geopandas, google-cloud-bigquery, h5netcdf, holoviews, jupyter-server, matplotlib, nbconvert, pooch, pytest, scikit-image, spacy, Sphinx, statsmodels, tables, tensorflow, thinc, xarray


In [3]:
pip install "packaging<=21.3"

In [4]:
from packaging.version import Version

version_string = "21.3"
version = Version(version_string)

print(version)

21.3




*   하단의 코드에서 패키징 버전이 23.1이 아닌 **21.3 이하 버전**일 때 pycaret import 가능

In [5]:
import packaging

print(packaging.__version__)

21.3


pycaret 설치 오류 관련 링크 (버전 문제) <br/>
https://coding-potato.tistory.com/7

pycaret 동작 방법 <br/>
https://www.notion.so/pycaret-836baab165694d519ad90fd8c1dc6713?pvs=4

In [6]:
#pycaret 돌릴 때 필요한 코드, 단 처음 돌린 후 런타임 재시작하고 다시 돌려야 돌아가는 단점 존재
!pip install -U pypandoc
!pip install  pycaret[full]
#!pip install markupsafe==2.1.3

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 484.4/484.4 kB 5.8 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 153.4/153.4 kB 9.8 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 81.9/81.9 kB 11.7 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 212.5/212.5 kB 11.2 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 79.9/79.9 MB 7.8 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 106.8/106.8 kB 17.4 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 46.4/46.4 kB 7.2 MB/s eta 0:00:00
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 17.0/17.0 MB 81.9 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 44.0/44.0 kB 5.2 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.8/1.8 MB 69.2 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━

In [7]:
from pycaret.classification import *

In [8]:
import re
import pandas as pd
import numpy as np

import seaborn as sns
import matplotlib.pyplot as plt

In [9]:
import warnings
warnings.filterwarnings(action='ignore')

In [10]:
from sklearn.feature_extraction.text import TfidfVectorizer
from sklearn.linear_model import LogisticRegression

In [11]:
train_r = '/content/drive/MyDrive/Colab Notebooks/dacon/law contest/data/train.csv'
test_r= '/content/drive/MyDrive/Colab Notebooks/dacon/law contest/data/test.csv'
temp_r='/content/drive/MyDrive/Colab Notebooks/dacon/law contest/data/0702_0.5_train.csv'
tempT_r='/content/drive/MyDrive/Colab Notebooks/dacon/law contest/data/0702_0.5_test.csv'

In [12]:
train = pd.read_csv(train_r)
test = pd.read_csv(test_r)
new = pd.read_csv(temp_r)
newT=pd.read_csv(tempT_r)

In [13]:
new

first_party_ner_ORG  first_party_ner_PERSON  first_party_ner_other  \
0                       0                       1                      0   
1                       0                       1                      0   
2                       0                       1                      0   
3                       0                       0                      0   
4                       0                       1                      0   
...                   ...                     ...                    ...   
2473                    1                       0                      0   
2474                    0                       1                      0   
2475                    1                       0                      0   
2476                    1                       0                      0   
2477                    0                       0                      0   

      first_party_ner_others  second_party_ner_ORG  second_party_ner_PERSON  \
0                          0                     0                        1   
1                          0                     0                        1   
2                          0                     0                        0   
3                          1                     0                        0   
4                          0                     0                        0   
...                      ...                   ...                      ...   
2473                       0                     1                        0   
2474                       0                     1                        0   
2475                       0                     0                        0   
2476                       0                     0                        0   
2477                       1                     1                        0   

      second_party_ner_other  second_party_ner_others  issued_area_criminal  \
0                          0                        0                     0   
1                          0                        0                     1   
2                          0                        0                     1   
3                          0                        1                     0   
4                          0                        0                     0   
...                      ...                      ...                   ...   
2473                       0                        0                     0   
2474                       0                        0                     0   
2475                       0                        0                     1   
2476                       0                        1                     0   
2477                       0                        0                     0   

      issued_area_others  ... well   whether  within  without  woman  work  \
0                      1  ...  0.0  0.153772     0.0      0.0    0.0   0.0   
1                      0  ...  0.0  0.000000     0.0      0.0    0.0   0.0   
2                      0  ...  0.0  0.000000     0.0      0.0    0.0   0.0   
3                      1  ...  0.0  0.000000     0.0      0.0    0.0   0.0   
4                      1  ...  0.0  0.000000     0.0      0.0    0.0   0.0   
...                  ...  ...  ...       ...     ...      ...    ...   ...   
2473                   1  ...  0.0  0.000000     0.0      0.0    0.0   0.0   
2474                   1  ...  0.0  0.000000     0.0      0.0    0.0   0.0   
2475                   0  ...  0.0  0.000000     0.0      0.0    0.0   0.0   
2476                   1  ...  0.0  0.000000     0.0      0.0    0.0   0.0   
2477                   1  ...  0.0  0.000000     0.0      0.0    0.0   0.0   

         would      writ      year  york  
0     0.000000  0.000000  0.000000   0.0  
1     0.000000  0.284502  0.091068   0.0  
2     0.000000  0.000000  0.000000   0.0  
3     0.000000  0.000000  0.000000   0.0  
4     0.116072  0.000000  0.000000   0.0  


In [14]:
newT

first_party_ner_ORG  first_party_ner_PERSON  first_party_ner_other  \
0                       0                       0                      0   
1                       0                       1                      0   
2                       1                       0                      0   
3                       0                       1                      0   
4                       0                       1                      0   
...                   ...                     ...                    ...   
1235                    1                       0                      0   
1236                    0                       0                      0   
1237                    0                       1                      0   
1238                    1                       0                      0   
1239                    0                       1                      0   

      first_party_ner_others  second_party_ner_ORG  second_party_ner_PERSON  \
0                          1                     0                        0   
1                          0                     1                        0   
2                          0                     1                        0   
3                          0                     0                        0   
4                          0                     0                        0   
...                      ...                   ...                      ...   
1235                       0                     1                        0   
1236                       1                     1                        0   
1237                       0                     0                        1   
1238                       0                     0                        1   
1239                       0                     0                        0   

      second_party_ner_other  second_party_ner_others  issued_area_criminal  \
0                          0                        0                     0   
1                          0                        0                     1   
2                          0                        0                     1   
3                          0                        0                     0   
4                          0                        1                     0   
...                      ...                      ...                   ...   
1235                       0                        0                     1   
1236                       0                        0                     1   
1237                       0                        0                     0   
1238                       0                        0                     0   
1239                       0                        0                     0   

      issued_area_others  ...  well  whether  within   without  woman  work  \
0                      1  ...   0.0      0.0     0.0  0.000000    0.0   0.0   
1                      0  ...   0.0      0.0     0.0  0.000000    0.0   0.0   
2                      0  ...   0.0      0.0     0.0  0.146081    0.0   0.0   
3                      1  ...   0.0      0.0     0.0  0.000000    0.0   0.0   
4                      1  ...   0.0      0.0     0.0  0.000000    0.0   0.0   
...                  ...  ...   ...      ...     ...       ...    ...   ...   
1235                   0  ...   0.0      0.0     0.0  0.124133    0.0   0.0   
1236                   0  ...   0.0      0.0     0.0  0.000000    0.0   0.0   
1237                   1  ...   0.0      0.0     0.0  0.000000    0.0   0.0   
1238                   1  ...   0.0      0.0     0.0  0.000000    0.0   0.0   
1239                   1  ...   0.0      0.0     0.0  0.000000    0.0   0.0   

      would  writ  year      york  
0       0.0   0.0   0.0  0.000000  
1       0.0   0.0   0.0  0.000000  
2       0.0   0.0   0.0  0.000000  
3       0.0   0.0   0.0  0.000000  
4       0.0   0.0   0.0  0.000000  
...     ...   ...   ...       

In [ ]:
#new.drop(columns=['Unnamed: 0','first_party','second_party'],inplace=True)
#newT.drop(columns=['Unnamed: 0','first_party','second_party'],inplace=True)
#new

KeyError: ignored

In [15]:
newT['ID']=test['ID']
newT

first_party_ner_ORG  first_party_ner_PERSON  first_party_ner_other  \
0                       0                       0                      0   
1                       0                       1                      0   
2                       1                       0                      0   
3                       0                       1                      0   
4                       0                       1                      0   
...                   ...                     ...                    ...   
1235                    1                       0                      0   
1236                    0                       0                      0   
1237                    0                       1                      0   
1238                    1                       0                      0   
1239                    0                       1                      0   

      first_party_ner_others  second_party_ner_ORG  second_party_ner_PERSON  \
0                          1                     0                        0   
1                          0                     1                        0   
2                          0                     1                        0   
3                          0                     0                        0   
4                          0                     0                        0   
...                      ...                   ...                      ...   
1235                       0                     1                        0   
1236                       1                     1                        0   
1237                       0                     0                        1   
1238                       0                     0                        1   
1239                       0                     0                        0   

      second_party_ner_other  second_party_ner_others  issued_area_criminal  \
0                          0                        0                     0   
1                          0                        0                     1   
2                          0                        0                     1   
3                          0                        0                     0   
4                          0                        1                     0   
...                      ...                      ...                   ...   
1235                       0                        0                     1   
1236                       0                        0                     1   
1237                       0                        0                     0   
1238                       0                        0                     0   
1239                       0                        0                     0   

      issued_area_others  ...  whether  within   without  woman  work  would  \
0                      1  ...      0.0     0.0  0.000000    0.0   0.0    0.0   
1                      0  ...      0.0     0.0  0.000000    0.0   0.0    0.0   
2                      0  ...      0.0     0.0  0.146081    0.0   0.0    0.0   
3                      1  ...      0.0     0.0  0.000000    0.0   0.0    0.0   
4                      1  ...      0.0     0.0  0.000000    0.0   0.0    0.0   
...                  ...  ...      ...     ...       ...    ...   ...    ...   
1235                   0  ...      0.0     0.0  0.124133    0.0   0.0    0.0   
1236                   0  ...      0.0     0.0  0.000000    0.0   0.0    0.0   
1237                   1  ...      0.0     0.0  0.000000    0.0   0.0    0.0   
1238                   1  ...      0.0     0.0  0.000000    0.0   0.0    0.0   
1239                   1  ...      0.0     0.0  0.000000    0.0   0.0    0.0   

      writ  year      york         ID  
0      0.0   0.0  0.000000  TEST_0000  
1      0.0   0.0  0.000000  TEST_0001  
2      0.0   0.0  0.000000  TEST_0002  
3      0.0   0.0  0.000000  TEST_0003  
4      0.0   0.0  0.000000  TEST_0

In [16]:
new=pd.merge(new,train)
newT=pd.merge(newT,test)

In [17]:
new

first_party_ner_ORG  first_party_ner_PERSON  first_party_ner_other  \
0                       0                       1                      0   
1                       0                       1                      0   
2                       0                       1                      0   
3                       0                       0                      0   
4                       0                       1                      0   
...                   ...                     ...                    ...   
2473                    1                       0                      0   
2474                    0                       1                      0   
2475                    1                       0                      0   
2476                    1                       0                      0   
2477                    0                       0                      0   

      first_party_ner_others  second_party_ner_ORG  second_party_ner_PERSON  \
0                          0                     0                        1   
1                          0                     0                        1   
2                          0                     0                        0   
3                          1                     0                        0   
4                          0                     0                        0   
...                      ...                   ...                      ...   
2473                       0                     1                        0   
2474                       0                     1                        0   
2475                       0                     0                        0   
2476                       0                     0                        0   
2477                       1                     1                        0   

      second_party_ner_other  second_party_ner_others  issued_area_criminal  \
0                          0                        0                     0   
1                          0                        0                     1   
2                          0                        0                     1   
3                          0                        1                     0   
4                          0                        0                     0   
...                      ...                      ...                   ...   
2473                       0                        0                     0   
2474                       0                        0                     0   
2475                       0                        0                     1   
2476                       0                        1                     0   
2477                       0                        0                     0   

      issued_area_others  ... without  woman  work     would      writ  \
0                      1  ...     0.0    0.0   0.0  0.000000  0.000000   
1                      0  ...     0.0    0.0   0.0  0.000000  0.284502   
2                      0  ...     0.0    0.0   0.0  0.000000  0.000000   
3                      1  ...     0.0    0.0   0.0  0.000000  0.000000   
4                      1  ...     0.0    0.0   0.0  0.116072  0.000000   
...                  ...  ...     ...    ...   ...       ...       ...   
2473                   1  ...     0.0    0.0   0.0  0.094845  0.000000   
2474                   1  ...     0.0    0.0   0.0  0.087753  0.000000   
2475                   0  ...     0.0    0.0   0.0  0.000000  0.000000   
2476                   1  ...     0.0    0.0   0.0  0.000000  0.000000   
2477                   1  ...     0.0    0.0   0.0  0.000000  0.000000   

          year  york                                   first_party  \
0     0.000000   0.0                             Phil A. St. Amant   
1     0.091068   0.0                                Stephen Duncan   
2     0.000000   0.0                             Billy Joe Magwood   
3     0.000000   0.0      

In [18]:
new.set_index('ID', drop = True, inplace=True)
newT.set_index('ID', drop = True, inplace=True)

In [19]:
new.drop(columns=['facts'],inplace=True)
newT.drop(columns=['facts'],inplace=True)

In [ ]:
new

first_party_ner_ORG  first_party_ner_PERSON  \
ID                                                        
TRAIN_0000                    0                       1   
TRAIN_0001                    0                       1   
TRAIN_0002                    0                       1   
TRAIN_0003                    0                       0   
TRAIN_0004                    0                       1   
...                         ...                     ...   
TRAIN_2473                    1                       0   
TRAIN_2474                    0                       1   
TRAIN_2475                    1                       0   
TRAIN_2476                    1                       0   
TRAIN_2477                    0                       0   

            first_party_ner_other  first_party_ner_others  \
ID                                                          
TRAIN_0000                      0                       0   
TRAIN_0001                      0                       0   
TRAIN_0002                      0                       0   
TRAIN_0003                      0                       1   
TRAIN_0004                      0                       0   
...                           ...                     ...   
TRAIN_2473                      0                       0   
TRAIN_2474                      0                       0   
TRAIN_2475                      0                       0   
TRAIN_2476                      0                       0   
TRAIN_2477                      0                       1   

            second_party_ner_ORG  second_party_ner_PERSON  \
ID                                                          
TRAIN_0000                     0                        1   
TRAIN_0001                     0                        1   
TRAIN_0002                     0                        0   
TRAIN_0003                     0                        0   
TRAIN_0004                     0                        0   
...                          ...                      ...   
TRAIN_2473                     1                        0   
TRAIN_2474                     1                        0   
TRAIN_2475                     0                        0   
TRAIN_2476                     0                        0   
TRAIN_2477                     1                        0   

            second_party_ner_other  second_party_ner_others  \
ID                                                            
TRAIN_0000                       0                        0   
TRAIN_0001                       0                        0   
TRAIN_0002                       0                        0   
TRAIN_0003                       0                        1   
TRAIN_0004                       0                        0   
...                            ...                      ...   
TRAIN_2473                       0                        0   
TRAIN_2474                       0                        0   
TRAIN_2475                       0                        0   
TRAIN_2476                       0                        1   
TRAIN_2477                       0                        0   

            issued_area_criminal  issued_area_others  ...  well   whether  \
ID                                                    ...                   
TRAIN_0000                     0                   1  ...   0.0  0.153772   
TRAIN_0001                     1                   0  ...   0.0  0.000000   
TRAIN_0002                     1                   0  ...   0.0  0.000000   
TRAIN_0003                     0                   1  ...   0.0  0.000000   
TRAIN_0004                     0                   1  ...   0.0  0.000000   
...                          ...                 ...  ...   ...       ...   
TRAIN_2473                     0                   1  ...   0.0  0.000000   
TRAIN_2474                     0                   1  ...   0.0  0.000000   
TRAIN_2475                     1                   0  ...   0.0  0

In [20]:
newT

first_party_ner_ORG  first_party_ner_PERSON  first_party_ner_other  \
ID                                                                              
TEST_0000                    0                       0                      0   
TEST_0001                    0                       1                      0   
TEST_0002                    1                       0                      0   
TEST_0003                    0                       1                      0   
TEST_0004                    0                       1                      0   
...                        ...                     ...                    ...   
TEST_1235                    1                       0                      0   
TEST_1236                    0                       0                      0   
TEST_1237                    0                       1                      0   
TEST_1238                    1                       0                      0   
TEST_1239                    0                       1                      0   

           first_party_ner_others  second_party_ner_ORG  \
ID                                                        
TEST_0000                       1                     0   
TEST_0001                       0                     1   
TEST_0002                       0                     1   
TEST_0003                       0                     0   
TEST_0004                       0                     0   
...                           ...                   ...   
TEST_1235                       0                     1   
TEST_1236                       1                     1   
TEST_1237                       0                     0   
TEST_1238                       0                     0   
TEST_1239                       0                     0   

           second_party_ner_PERSON  second_party_ner_other  \
ID                                                           
TEST_0000                        0                       0   
TEST_0001                        0                       0   
TEST_0002                        0                       0   
TEST_0003                        0                       0   
TEST_0004                        0                       0   
...                            ...                     ...   
TEST_1235                        0                       0   
TEST_1236                        0                       0   
TEST_1237                        1                       0   
TEST_1238                        1                       0   
TEST_1239                        0                       0   

           second_party_ner_others  issued_area_criminal  issued_area_others  \
ID                                                                             
TEST_0000                        0                     0                   1   
TEST_0001                        0                     1                   0   
TEST_0002                        0                     1                   0   
TEST_0003                        0                     0                   1   
TEST_0004                        1                     0                   1   
...                            ...                   ...                 ...   
TEST_1235                        0                     1                   0   
TEST_1236                        0                     1                   0   
TEST_1237                        0                     0                   1   
TEST_1238                        0                     0                   1   
TEST_1239                        0                     0                   1   

           ...  within   without  woman  work  would  writ  year      york  \
ID         ...                                                               
TEST_0000  ...     0.0  0.000000    0.0   0.0    0.0   0.0   0.0  0.000000   
TEST_0001  ...     0.0  0.000000    0.0   0.0    0.0   0.0   0.0  0.000000   
TEST_0002  ...     0.0  0.146

In [21]:
#fix_imbalance=True를 통해 SMOTE를 적용하여 클래스가 적은 데이터의 표본을 임의 값으로 변형하여 데이터 추가 적용
exp=setup(data=new, target='first_party_winner', session_id=4454, fix_imbalance=True, normalize=True)

In [22]:
 best_15 = compare_models(sort='Accuracy', n_select=15)

Processing:   0%|          | 0/83 [00:00<?, ?it/s]


### ✔ Gradient Boosting Classifier
*   **Predict 0 Result** </br>
train : 684 </br>
test  : 181 </br> </br>

acuuracy : 0.6534

In [43]:
%time tuned_model = tune_model(best_15[2],choose_better=True, optimize='Accuracy')

Processing:   0%|          | 0/7 [00:00<?, ?it/s]

Fitting 10 folds for each of 10 candidates, totalling 100 fits


Original model was better than the tuned model, hence it will be returned. NOTE: The display metrics are for the tuned model (not the original one).
CPU times: user 11.9 s, sys: 5.96 s, total: 17.8 s
Wall time: 2min 17s


In [59]:
tuned_model

GradientBoostingClassifier(ccp_alpha=0.0, criterion='friedman_mse', init=None,
                           learning_rate=0.1, loss='log_loss', max_depth=3,
                           max_features=None, max_leaf_nodes=None,
                           min_impurity_decrease=0.0, min_samples_leaf=1,
                           min_samples_split=2, min_weight_fraction_leaf=0.0,
                           n_estimators=100, n_iter_no_change=None,
                           random_state=4454, subsample=1.0, tol=0.0001,
                           validation_fraction=0.1, verbose=0,
                           warm_start=False)

In [44]:
predictions = predict_model(tuned_model, data=new)
y_pred=predictions.prediction_label

In [45]:
y_pred.value_counts()

1    1794
0     684
Name: prediction_label, dtype: int64

In [46]:
predictions = predict_model(tuned_model, data=newT)
y_pred = predictions.prediction_label

In [47]:
y_pred.value_counts()

1    1059
0     181
Name: prediction_label, dtype: int64

In [48]:
y_pred=y_pred.to_frame()

In [49]:
y_pred

prediction_label
ID                         
TEST_0000                 0
TEST_0001                 1
TEST_0002                 1
TEST_0003                 0
TEST_0004                 1
...                     ...
TEST_1235                 1
TEST_1236                 1
TEST_1237                 1
TEST_1238                 1
TEST_1239                 1

[1240 rows x 1 columns]

In [50]:
index_list = list(range(1240))

In [51]:
index_list=[index_list]

In [52]:
index_list

[[0,
  1,
  2,
  3,
  4,
  5,
  6,
  7,
  8,
  9,
  10,
  11,
  12,
  13,
  14,
  15,
  16,
  17,
  18,
  19,
  20,
  21,
  22,
  23,
  24,
  25,
  26,
  27,
  28,
  29,
  30,
  31,
  32,
  33,
  34,
  35,
  36,
  37,
  38,
  39,
  40,
  41,
  42,
  43,
  44,
  45,
  46,
  47,
  48,
  49,
  50,
  51,
  52,
  53,
  54,
  55,
  56,
  57,
  58,
  59,
  60,
  61,
  62,
  63,
  64,
  65,
  66,
  67,
  68,
  69,
  70,
  71,
  72,
  73,
  74,
  75,
  76,
  77,
  78,
  79,
  80,
  81,
  82,
  83,
  84,
  85,
  86,
  87,
  88,
  89,
  90,
  91,
  92,
  93,
  94,
  95,
  96,
  97,
  98,
  99,
  100,
  101,
  102,
  103,
  104,
  105,
  106,
  107,
  108,
  109,
  110,
  111,
  112,
  113,
  114,
  115,
  116,
  117,
  118,
  119,
  120,
  121,
  122,
  123,
  124,
  125,
  126,
  127,
  128,
  129,
  130,
  131,
  132,
  133,
  134,
  135,
  136,
  137,
  138,
  139,
  140,
  141,
  142,
  143,
  144,
  145,
  146,
  147,
  148,
  149,
  150,
  151,
  152,
  153,
  154,
  155,
  156,
  157,
  15

In [53]:
y_pred = y_pred.set_index(keys=index_list, inplace=False, drop=False)

In [54]:
submit = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/dacon/law contest/data/sample_submission.csv')

In [55]:
submit

ID  first_party_winner
0     TEST_0000                   0
1     TEST_0001                   0
2     TEST_0002                   0
3     TEST_0003                   0
4     TEST_0004                   0
...         ...                 ...
1235  TEST_1235                   0
1236  TEST_1236                   0
1237  TEST_1237                   0
1238  TEST_1238                   0
1239  TEST_1239                   0

[1240 rows x 2 columns]

In [56]:
submit['first_party_winner'] = y_pred['prediction_label']

In [57]:
submit

ID  first_party_winner
0     TEST_0000                   0
1     TEST_0001                   1
2     TEST_0002                   1
3     TEST_0003                   0
4     TEST_0004                   1
...         ...                 ...
1235  TEST_1235                   1
1236  TEST_1236                   1
1237  TEST_1237                   1
1238  TEST_1238                   1
1239  TEST_1239                   1

[1240 rows x 2 columns]

In [58]:
submit.to_csv('./pycaretGBC.csv', index=False)
print('Done')

Done



### ✔ Light Gradient Boosting Machine
*   **Predict 0 Result** </br>
train : 644 </br>
test  : 118 </br> </br>

acuuracy : 0.6569

In [33]:
%time tuned_model = tune_model(best_15[5],choose_better=True, optimize='Accuracy')

Processing:   0%|          | 0/7 [00:00<?, ?it/s]

Fitting 10 folds for each of 10 candidates, totalling 100 fits


CPU times: user 12.4 s, sys: 4.26 s, total: 16.6 s
Wall time: 2min 53s


In [34]:
predictions = predict_model(tuned_model, data=new)
y_pred=predictions.prediction_label

In [35]:
y_pred.value_counts()

1    1834
0     644
Name: prediction_label, dtype: int64

In [36]:
predictions = predict_model(tuned_model, data=newT)
y_pred = predictions.prediction_label

In [37]:
y_pred.value_counts()

1    1122
0     118
Name: prediction_label, dtype: int64

In [ ]:
y_pred

prediction_label
0                    0
1                    0
2                    1
3                    1
4                    1
...                ...
1235                 1
1236                 0
1237                 1
1238                 1
1239                 1

[1240 rows x 1 columns]

In [ ]:
tuned_model


### ✔ Random Forest
*   **Predict '0' Result** </br>
train : 704 </br>
test  : 198 </br></br>

accuracy : 0.5999

In [38]:
%time tuned_model = tune_model(best_15[6],choose_better=True, optimize='Accuracy')

Processing:   0%|          | 0/7 [00:00<?, ?it/s]

Fitting 10 folds for each of 10 candidates, totalling 100 fits


Original model was better than the tuned model, hence it will be returned. NOTE: The display metrics are for the tuned model (not the original one).
CPU times: user 12.1 s, sys: 5.43 s, total: 17.5 s
Wall time: 5min 31s


In [39]:
predictions = predict_model(tuned_model, data=new)
y_pred=predictions.prediction_label

In [40]:
y_pred.value_counts()

1    1774
0     704
Name: prediction_label, dtype: int64

In [41]:
predictions = predict_model(tuned_model, data=newT)
y_pred = predictions.prediction_label

In [42]:
y_pred.value_counts()

1    1042
0     198
Name: prediction_label, dtype: int64


### ✔ Extra Trees Classifier
*   **Predict 0 Result** </br>
train : 639 </br>
test  : 130 </br></br>

accuracy : 0.5553


In [ ]:
tuned_model = tune_model(best_15[2])

Processing:   0%|          | 0/7 [00:00<?, ?it/s]

Fitting 10 folds for each of 10 candidates, totalling 100 fits


Original model was better than the tuned model, hence it will be returned. NOTE: The display metrics are for the tuned model (not the original one).


In [ ]:
predictions = predict_model(tuned_model, data=new)
y_pred=predictions.prediction_label

In [ ]:
y_pred.value_counts()

1    1782
0     696
Name: prediction_label, dtype: int64

In [ ]:
predictions = predict_model(tuned_model, data=newT)
y_pred = predictions.prediction_label

In [ ]:
y_pred.value_counts()

1    1057
0     183
Name: prediction_label, dtype: int64


### ✔ Naive Bayes
*   **Predict 0 Result** </br>
train : 696 </br>
test  : 556 </br></br>

accuracy :  0.5325


In [ ]:
tuned_model = tune_model(best_15[2])

Processing:   0%|          | 0/7 [00:00<?, ?it/s]

Fitting 10 folds for each of 10 candidates, totalling 100 fits


In [ ]:
predictions = predict_model(tuned_model, data=new)
y_pred=predictions.prediction_label

In [ ]:
y_pred.value_counts()

0    933
1    696
Name: prediction_label, dtype: int64

In [ ]:
predictions = predict_model(tuned_model, data=newT)
y_pred = predictions.prediction_label

In [ ]:
y_pred.value_counts()

0    684
1    556
Name: prediction_label, dtype: int64



---




### ✔ XGBoost
*   **Predict 0 Result** </br>
train : 625 </br>
test  : 103 </br>

In [ ]:
tuned_model = tune_model(best_15[3])

Processing:   0%|          | 0/7 [00:00<?, ?it/s]

Fitting 10 folds for each of 10 candidates, totalling 100 fits


In [ ]:
predictions = predict_model(tuned_model, data=new)
y_pred=predictions.prediction_label

In [ ]:
y_pred.value_counts()

1    1853
0     625
Name: prediction_label, dtype: int64

In [ ]:
predictions = predict_model(tuned_model, data=newT)
y_pred = predictions.prediction_label

In [ ]:
y_pred.value_counts()

1    1137
0     103
Name: prediction_label, dtype: int64


### ✔ CatBoost
*   **Predict 0 Result** </br>
train : 627 </br>
test  : 97 </br>

*이것만 돌아가는데 한 시간 20분 걸렸어요..

In [ ]:
tuned_model = tune_model(best_15[4])

Processing:   0%|          | 0/7 [00:00<?, ?it/s]

Fitting 10 folds for each of 10 candidates, totalling 100 fits


In [ ]:
predictions = predict_model(tuned_model, data=new)
y_pred=predictions.prediction_label

In [ ]:
y_pred.value_counts()

1    1851
0     627
Name: prediction_label, dtype: int64

In [ ]:
predictions = predict_model(tuned_model, data=newT)
y_pred = predictions.prediction_label

In [ ]:
y_pred.value_counts()

1    1143
0      97
Name: prediction_label, dtype: int64

In [ ]:
tuned_model

In [ ]:
submit = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/dacon/law contest/data/sample_submission.csv')
submit['first_party_winner'] = y_pred
submit.to_csv('./proces_tf_catboost.csv', index=False)
print('Done')

Done



### ✔ LGBM
*   **Predict 0 Result** </br>
train : 689 </br>
test  : 181 </br>

In [ ]:
tuned_model = tune_model(best_15[5])

Processing:   0%|          | 0/7 [00:00<?, ?it/s]

Fitting 10 folds for each of 10 candidates, totalling 100 fits


In [ ]:
predictions = predict_model(tuned_model, data=new)
y_pred=predictions.prediction_label

In [ ]:
y_pred.value_counts()

1    1789
0     689
Name: prediction_label, dtype: int64

In [ ]:
predictions = predict_model(tuned_model, data=newT)
y_pred = predictions.prediction_label

In [ ]:
y_pred.value_counts()

1    1059
0     181
Name: prediction_label, dtype: int64


### ✔ Naive Bayes
*   **Predict 0 Result** </br>
train : 1126 </br>
test  : 540 </br>

*이건 너무 많지 않나..?

In [ ]:
tuned_model = tune_model(best_15[13])

Processing:   0%|          | 0/7 [00:00<?, ?it/s]

Fitting 10 folds for each of 10 candidates, totalling 100 fits


Original model was better than the tuned model, hence it will be returned. NOTE: The display metrics are for the tuned model (not the original one).


In [ ]:
predictions = predict_model(tuned_model, data=new)
y_pred=predictions.prediction_label

In [ ]:
y_pred.value_counts()

1    1352
0    1126
Name: prediction_label, dtype: int64

In [ ]:
predictions = predict_model(tuned_model, data=newT)
y_pred = predictions.prediction_label

In [ ]:
y_pred.value_counts()

1    700
0    540
Name: prediction_label, dtype: int64

In [ ]:
submit = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/dacon/law contest/data/sample_submission.csv')

In [ ]:
submit['first_party_winner'] = y_pred

In [ ]:
submit.to_csv('./proces_tf_naive.csv', index=False)
print('Done')

Done


In [ ]:
#blended_l = blend_models(estimator_list = best_5, fold = 5, optimize = 'Accuracy')
#pred_holdout = predict_model(blended_l)
#final_model_l = finalize_model(blended_l)

Processing:   0%|          | 0/6 [00:00<?, ?it/s]